sys/vtimer/vtimer.c

```diff
@@ -134,8 +135,9 @@ static int update_shortterm(void)
         next = now +  HWTIMER_TICKS_TO_US(VTIMER_BACKOFF);
     }
 
-    DEBUG("update_shortterm: Set hwtimer to %" PRIu32 " (now=%lu)\n", next, HWTIMER_TICKS_TO_US(hwtimer_now()));
-    hwtimer_id = hwtimer_set_absolute(HWTIMER_TICKS(next), vtimer_callback, NULL);
+    DEBUG("update_shortterm: Set hwtimer to %" PRIu32 " (now=%lu)\n", next, now);
+    uint32_t next_ticks = now_ticks + HWTIMER_TICKS(next - now);
+    hwtimer_id = hwtimer_set_absolute(next_ticks, vtimer_callback, NULL);
```

Comment:
I don't really understand why the original implementation used `hwtimer_set_absolute` since they are setting a relative time. `hwtimer_set` is semantically more correct for this operation.